src/main/java/uk/gov/mca/beacons/service/beacons/BeaconsService.java

```diff
@@ -69,4 +74,30 @@ public Beacon find(UUID id) {
       beaconUses
     );
   }
+
+  public void update(UUID id, Beacon update) {
+    final Beacon beacon = this.find(id);
```

Comment:
@tm1303 As we're just updating beacon information here, there isn't a need to use the find method in this class.  You could look to use the following method: 

```java
final Optional<Beacon> beacon = beaconRepository.findById(id);
if (beacon.isEmpty()) throw new ResourceNotFoundException();
```
Which returns an optional if you cannot find the beacon and reads nicer if the exception is thrown

Reply:
you fell in to my trap 😈 

to complete the updated the model has to have something in the owner and uses properties. So whilst the above will work it will later fail when you reach the `beaconRepository.save(updatedModel);`.

Options: 
* Do as you suggest, but then populate the owner and uses with empty placeholders. This feels like it is setting us up for a fail later though (if a change to the update call starts managing relationships for example)
* Remove validation from the model. This has been suggested by Stu before for other reasons, but feels like its own TD ticket.